index.js

```diff
@@ -27,11 +27,14 @@ const decorateErrorWithCounts = (error, attemptNumber, options) => {
 	return error;
 };
 
+const shouldRetry = error => !(error instanceof TypeError) || isNetworkError(error);
```

Comment:
The isnetwork check is already done. Not sure what the point of doing it here is.

Reply:
What do you think about using the default return value for the function `shouldRetry` to be `true`?
```javascript
const shouldRetry = () => true;
```

Even if you check whether it is a TypeError here, you can only experience network errors because you have checked it before.